### PR TITLE
Kill pickling

### PR DESCRIPTION
**Pull request by vkostyukov**
Lets kill pickling!

We can merged it and move forward. I would like to start working on core tests. Do you mind me do that?

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -44,8 +43,9 @@ lazy val benchmark = project.in(file("serial-benchmark"))
   .settings(moduleName := "finagle-serial-bencrhmark")
```
Comment (travisbrown) [inline build.sbt]: Mind hitting the typo here ("bencrhmark") before merging? Otherwise looks good to me.

Comment (vkostyukov) [inline build.sbt]: BTW, what do you think about avoiding `finagle-serial` prefix in folders names but use it for maven?

**Comment by travisbrown** [comment]
There's not a lot we can test in core (because there's not really a lot in core), but I think it might be a good idea to add a `test` package for testing generic `Serial` implementations, so we don't duplicate a lot of work in the individual wrappers.

**Comment by vkostyukov** [comment]
Merging this.
